index.js

```diff
@@ -113,32 +123,45 @@ function text (scanner) {
     }
     node.value += scanner.next()
   }
+  node.position = { start, end: scanner.position() }
   return node
 }
 
 /*
  * <summary-sep>  ::= "!"? ":" *<whitespace>
  */
 function summarySep (scanner) {
+  const start = scanner.position()
   const node = {
     type: 'summary-sep',
     children: []
   }
   if (isSummarySep(scanner.peek())) {
     scanner.next()
+    // manually offset the end with half the "!:" size
+    const breakingEnd = scanner.position()
```

Comment:
It's a bit unfortunate, but I don't see a better way to make this happen.

Reply:
I'm wondering if we could drop the summary-sep node, and just make `!` a special case where we check whether the next node is a `:`, and then add the additional node if needed?